contracts/StablecoinConverter.sol

```diff
@@ -133,13 +137,19 @@ contract StablecoinConverter is EpochTokenLocker {
         uint128[] memory volumes,
         uint128[] memory prices,  //list of prices for touched token only
         uint16[] memory tokenIdsForPrice  // price[i] is the price for the token with tokenID tokenIdsForPrice[i]
+                                          // fee token id not required since always 0
     ) public {
         require(
             batchIndex == getCurrentStateIndex() - 1,
             "Solutions are no longer accepted for this batch"
         );
         require(checkPriceOrdering(tokenIdsForPrice), "prices are not ordered by tokenId");
-        undoPreviousSolution(batchIndex);
+        if (previousSolution.batchId == batchIndex) {
+            undoPreviousSolution();
+        } else {
+            currentFeeCollected = 0;
```

Comment:
Shouldn't this logic be part of `checkForBestSolutionSubmission` given that it will be overridden in the same method anyways? Seems a bit dislocated here.